controllers/storagecluster/cephblockpools.go

```diff
@@ -61,6 +61,7 @@ func (r *StorageClusterReconciler) newCephBlockPoolInstances(initData *ocsv1.Sto
 			},
 			Spec: cephv1.NamedBlockPoolSpec{
 				PoolSpec: cephv1.PoolSpec{
+					DeviceClass:    generateDeviceClass(initData),
```

Comment:
I don't think this needs to be a function at all. There's no reason to set the DeviceClass for the replica 3 block pools. Or at least, so I assume. @travisn should probably confirm. Really this ties into the fact that we don't want a global boolean for all CephBlockPools, so it can just be removed entirely.

Reply:
This is the one place where there is some side effect on resilient pools if the replica 1 feature is enabled... The `deviceClass: replicated` is needed on all of the resilient pools so that Ceph can differentiate placing data on different sets of OSDs. If we didn't add this property to the resilient pools, the resilient pools would place data on the OSDs that are meant to be reserved for replica 1 pools.

Reply:
Ah-ha... okay. That seems a little odd, but makes sense.